### ISSUE TITLE
Lint comments after backslashes for line continuations

### ISSUE THREAD

**Issue by keith** [koalaman/shellcheck#2438]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related (but I'm surprised I didn't)

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

echo this \
  works

echo foo \
  # Some comment
  bar

echo foo \ # Some comment
  bar
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Some warning about how neither of these 2 comments are doing what the user would expect, and instead likely end up breaking the multi-line-string the user was trying to create.
